tests/handlers/records-read.spec.ts

```diff
@@ -1614,6 +1917,49 @@ export function testRecordsReadHandler(): void {
       });
     });
 
+    describe('createFilter()', async () => {
+      it('should not throw if only `recordId` is set', async () => {
+        const filter = RecordsReadHandler.createFilter({
+          interface        : DwnInterfaceName.Records,
+          method           : DwnMethodName.Read,
+          messageTimestamp : '2023-08-19T00:00:00.000000Z',
+
+          recordId: 'some-id'
+        });
+
+        expect(filter['recordId']).to.equal('some-id');
+      });
+
+      it('should not throw if `protocol` and `protocolPath` are set', async () => {
+        const filter = RecordsReadHandler.createFilter({
+          interface        : DwnInterfaceName.Records,
+          method           : DwnMethodName.Read,
+          messageTimestamp : '2023-08-19T00:00:00.000000Z',
+
+          protocol     : 'some-protocol',
+          protocolPath : 'protocol/path'
+        });
+
+        expect(filter['protocol']).to.equal('some-protocol');
+        expect(filter['protocolPath']).to.equal('protocol/path');
+      });
+
+      it('should not set protocol filters if `recordId` is provided', async () => {
+        const filter = RecordsReadHandler.createFilter({
+          interface        : DwnInterfaceName.Records,
+          method           : DwnMethodName.Read,
+          messageTimestamp : '2023-08-19T00:00:00.000000Z',
+
+          protocol     : 'some-protocol',
+          protocolPath : 'protocol/path',
+          recordId     : 'some-id'
+        });
+
+        expect(filter['recordId']).to.equal('some-id');
+        expect(filter['protocol']).to.be.undefined;
+        expect(filter['protocolPath']).to.be.undefined;
```

Comment:
Any reason why you're using `['']` notation instead of dot notation here? I prefer dot notation since the typechecker isn't good with square bracket notation.

Reply:
Filter's type is generalized to `[property: string]: ...` so there is no strong typing of any of the filter properties.

Reply:
Filter's type is `[property: string]: EqualFilter | OneOfFilter | RangeFilter`, so there is typing of the value. For example, the following does not build because `{ bar: { baz: 1 } }` is not a `EqualFilter | OneOfFilter | RangeFilter`

```ts
const filter: Filter = {
  foo: { bar: { baz: 1 } },
};
```